src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractDatabaseEngine.java

```diff
@@ -272,6 +283,9 @@ protected void connect() throws Exception {
 
         this.conn = DriverManager.getConnection(jdbc, props);
 
+        // the "network timeout" is specified in milliseconds, needs to be converted from seconds
+        this.conn.setNetworkTimeout(socketTimeoutExecutor, this.properties.getSocketTimeout() * 1000);
```

Comment:
fair point...
I tried this, but as strange as it may seem, the method signature is
`    void setNetworkTimeout(Executor executor, int milliseconds) throws SQLException;`
it needs milliseconds, but _int_ :man_shrugging: